packages/grafana-data/src/themes/createColors.ts

```diff
@@ -0,0 +1,200 @@
+import { merge } from 'lodash';
+import { palette } from './palette';
+import { DeepPartial } from './types';
+
+export interface ThemeRichColor {
+  /** color intent (primary, secondary, info, error, etc) */
+  name: string;
+  /** Main color */
+  main: string;
+  /** Light shade */
+  light: string;
+  /** Dark shade */
+  dark: string;
+  /** Text color for text ontop of main */
+  contrastText: string;
+}
```

Comment:
this part is very unknown still, will we need light & dark shades? they cannot be used in the same way in dark and light theme (For the most part). 
Alternative would be 
* main (for backgrounds)
* borderAndText (for text & borders, not sure if this can be the same variable but probably) 
* transparent (for subtly colored backgrounds based on main, might be used for alerts depending on the final design for alerts) 
* contrastText (For the text to use for text placed on main, ie button color) 

Hover color can use the function. thoughts @Ijin08